lib/Server.js

```diff
@@ -405,19 +406,17 @@ Server.prototype.checkHost = function(headers) {
 	// we don't care about port not matching
 	const hostHeader = headers.host;
 	if(!hostHeader) return false;
-	const idx = hostHeader.indexOf(":");
-	const hostname = idx >= 0 ? hostHeader.substr(0, idx) : hostHeader;
+	const hostname = url.parse(hostHeader).hostname;
 
 	// always allow localhost host, for convience
-	if(hostname === "127.0.0.1" || hostname === "localhost") return true;
+	if(hostname === "127.0.0.1" || hostname === "localhost" || hostname === "::1") return true;
 
 	// allow hostname of listening adress
 	if(hostname === this.listenHostname) return true;
 
 	// also allow public hostname if provided
 	if(typeof this.publicHost === "string") {
```

Comment:
It would be great if `publicHost` could be an array so we could provide an IPv6 and IPv4 address.